packages/roosterjs-editor-plugins/lib/plugins/CustomReplace/CustomReplace.ts

```diff
@@ -90,14 +96,23 @@ export default class CustomReplacePlugin implements EditorPlugin {
         if (range == null) {
             return;
         }
-        const searcher = this.editor.getContentSearcherOfCursor(event);
-        const stringToSearch = searcher.getSubStringBefore(this.longestReplacementLength);
 
+        const searcher = this.editor.getContentSearcherOfCursor(event);
+        const stringToSearch = searcher
+            .getSubStringBefore(this.longestReplacementLength)
+            .replace(/\s/g, ' ');
         const replacement = this.getMatchingReplacement(stringToSearch);
         if (replacement == null) {
             return;
         }
 
+        if (
+            replacement.shouldReplace &&
+            !replacement.shouldReplace(searcher.getWordBefore(), replacement.sourceString)
```

Comment:
Can we just pass replacement as the first parameter here?

Reply:
Open question: what about pass the whole replacement object as the second parameter?